core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -357,180 +360,45 @@ private void cleanExpiredSnapshots() {
 
     LOG.info("Committed snapshot changes; cleaning up expired manifests and data files.");
 
-    removeExpiredFiles(current.snapshots(), validIds, expiredIds);
+    removeExpiredFiles(base, current);
```

Comment:
What we may want to do is have 2 "strategies":

1.) If there is only a single branch, apply incremental expiration.

2.) If there are multiple branches, apply the reachability analysis. 

That way we get the benefits of incremental expiration when it's applicable.